src/function/maybe_changed_after.rs

```diff
@@ -252,7 +252,9 @@ where
     ) -> bool {
         // Wouldn't it be nice if rust had an implication operator ...
         // may_be_provisional -> validate_provisional
-        !memo.may_be_provisional() || self.validate_provisional(db, zalsa, database_key_index, memo)
+        !memo.may_be_provisional()
```

Comment:
The doc comment for this method (above and unchanged in the PR, so I can't inline-comment on it directly) needs to be updated to mention the same-iteration caveat.